src/corefx/System.Globalization.Native/pal_calendarData.c

```diff
@@ -572,17 +565,20 @@ GetLatestJapaneseEra
 
 Gets the latest era in the Japanese calendar.
 */
-extern "C" int32_t GlobalizationNative_GetLatestJapaneseEra()
+int32_t GlobalizationNative_GetLatestJapaneseEra()
 {
     UErrorCode err = U_ZERO_ERROR;
-    UCalendar* pCal = ucal_open(nullptr, 0, JAPANESE_LOCALE_AND_CALENDAR, UCAL_TRADITIONAL, &err);
-    UCalendarHolder calHolder(pCal, err);
+    UCalendar* pCal = ucal_open(NULL, 0, JAPANESE_LOCALE_AND_CALENDAR, UCAL_TRADITIONAL, &err);
 
     if (U_FAILURE(err))
+    {
+        ucal_close(pCal);
```

Comment:
In case of failure, the pCal should not be closed (see the original holder). That's why the holder gets the err as argument too.
The same at many other places below.